images/virtualization-artifact/pkg/controller/service/size_policy_service.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/deckhouse/virtualization/api/core/v1alpha2"
+)
+
+const (
+	lesser  = -1
+	equal   = 0
+	greater = 1
+)
+
+type SizePolicyService struct {
+	client client.Client
+}
+
+func NewSizePolicyService(client client.Client) *SizePolicyService {
+	return &SizePolicyService{client: client}
+}
+
+func (s *SizePolicyService) CheckVMMatchedSizePolicy(ctx context.Context, vm *v1alpha2.VirtualMachine) error {
+	vmclass := &v1alpha2.VirtualMachineClass{}
+	err := s.client.Get(ctx, types.NamespacedName{
+		Name: vm.Spec.VirtualMachineClassName,
+	}, vmclass)
+	if err != nil {
+		return err
+	}
+
+	sizePolicy := getVMSizePolicy(vm, vmclass)
+	if sizePolicy == nil {
+		return fmt.Errorf(
+			"virtual machine %s resources not match any of sizing policies in vm class %s",
```

Comment:
It's better to avoid showing users short names of resources: vm class -> virtual machine class ot VirtualMachineClass. 
Also use %q to refer to a name of resource.